component/src/main/ballerina/cassandra/actions.bal

```diff
@@ -0,0 +1,121 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+documentation {
+    The Datatype of the parameter.
+}
+public type Type
+"INT"|"BIGINT"|"VARINT"|"FLOAT"|"DOUBLE"|"TEXT"|"BOOLEAN"|"LIST";
+
+documentation {
+    A 32-bit signed integer.
+}
+@final public Type TYPE_INT = "INT";
+
+documentation {
+    A 64-bit signed long.
+}
+@final public Type TYPE_BIGINT = "BIGINT";
+
+documentation {
+    Arbitrary precision integer.
+}
+@final public Type TYPE_VARINT = "VARINT";
+
+documentation {
+     A 32-bit IEEE-754 floating point.
+}
+@final public Type TYPE_FLOAT = "FLOAT";
+
+documentation {
+    A 64-bit IEEE-754 floating point.
+}
+@final public Type TYPE_DOUBLE = "DOUBLE";
+
+documentation {
+    UTF-8 encoded string.
+}
+@final public Type TYPE_TEXT = "TEXT";
+
+documentation {
+    Boolean value either True or false.
+}
+@final public Type TYPE_BOOLEAN = "BOOLEAN";
+
+documentation {
+    A collection of one or more ordered elements.
+}
+@final public Type TYPE_LIST = "LIST";
+
+documentation {
+    Represents complex parameter passed to `select` or `update` operation.
+
+    F{{cqlType}} Cassandra type of the parameter
+    F{{value}} Value of the parameter
+}
+public type Parameter {
+    Type cqlType,
+    any value,
+};
+
+documentation {
+    Tuple type representing either a `Parameter` or a primitive ballerina type.
+}
+public type Param string|int|boolean|float|Parameter;
+
+documentation {
+    The Client Connector for Cassandra database.
+}
+public type CallerActions object {
+
+documentation {
+    Select data from cassandra datasource.
+
+    P{{queryString}} Query to be executed
+    P{{recordType}} The Type result should be mapped to
+    P{{parameters}} Parameter array used with the given query
+    R{{}} `table` representing the result of the select action or `error` if an error occurs
+}
+public native function select (string queryString, typedesc recordType, Param... parameters)
+returns (table | error);
+
+documentation {
+    Execute update query on cassandra datasource.
+
+    P{{queryString}} Query to be executed
+    P{{parameters}} Parameter array used with the given query
+    R{{}} `nil` or `error` if an error occurs
+}
+public native function update (string queryString, Param... parameters) returns (error?);
+
+};
+
+documentation {
+    The close action implementation to shutdown the cassandra connections.
+}
+public native function close (CallerActions callerActions);
+
+documentation {
+    CassandraConnectorError type represents an error occured during the Cassandra client invocation.
+
+    F{{message}}  An error message explaining about the error
+    F{{cause}} The error(s) that caused CassandraConnectorError to get thrown
+}
+public type CassandraConnectorError {
+    string message,
+    error[] cause,
+};
```

Comment:
Use error instead of defining a new error type as we are not having any additional information than the error type